ui/src/SensitiveData.js

```diff
@@ -0,0 +1,161 @@
+      newValue: "",
+      newValueValidationError: "",
+      showDialog: false,
+    });
+  };
+
+  getSensitiveSampleColumns = async () => {
+    const response = await fetch(`/api/surveys/${this.props.surveyId}`);
+    if (!response.ok) {
+      return;
+    }
+
+    const surveyJson = await response.json();
+    const sensitiveColumns = surveyJson.sampleValidationRules
+      .filter((rule) => rule.sensitive)
+      .map((rule) => rule.columnName);
+
+    return sensitiveColumns;
+  };
+
+  onSensitiveDataColumnChange = (event) => {
+    this.setState({ columnToUpdate: event.target.value });
+  };
+
+  onChangeValue = (event) => {
+    this.setState({
+      newValue: event.target.value,
+    });
+  };
+
+  refreshDataFromBackend = async () => {
+    const sensitiveDataColumns = await this.getSensitiveSampleColumns(this.props.surveyId
+    ).then((sensitiveColumns) => {
+      return sensitiveColumns;
+    });
+
+    this.setState({
+      allowableSensitiveDataColumns: sensitiveDataColumns,
+    });
+  };
+
+  onCreate = async () => {
+    const updateSampleSensitive = {
+      caseId: this.props.caseId,
+      sampleSensitive: { [this.state.columnToUpdate]: this.state.newValue }
+    }
+
+    const response = await fetch(
+      `/api/cases/${this.props.caseId}/action/updateSensitiveField`,
+      {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(updateSampleSensitive),
+      }
+    );
+
+    if (response.ok) {
+      alert('Update msg sent successfully')
+      this.closeDialog();
+      return;
+    }
+
+    const data = await response.json();
+    this.setState({ newValueValidationError: data.errors });
+  };
+
+  render() {
+
+    const sensitiveDataColumnMenuItems =
+      this.state.allowableSensitiveDataColumns.map((columnName) => (
+        <MenuItem key={columnName} value={columnName}>
+          {columnName}
+        </MenuItem>
+      ));
+
+    return (
+      <div>
+        <Button onClick={this.openDialog} variant="contained" style={{ marginTop: 10 }}>
+          Modify Sensitive Data
+        </Button>
+        <Dialog open={this.state.showDialog}>
+          <DialogContent style={{ padding: 30 }}>
+            <div>
+              <FormControl required fullWidth={true}>
+                <InputLabel>Sensitive Data Columns</InputLabel>
+                <Select
+                  onChange={this.onSensitiveDataColumnChange}
+                  value={this.state.columnToUpdate}
+                  error={this.state.newValueValidationError}
+                >
+                  {sensitiveDataColumnMenuItems}
+                </Select>
+              </FormControl>
+              <TextField
+                required
+                style={{ minWidth: 200 }}
+                label="new value, blank is valid"
+                onChange={this.onChangeValue}
+                value={this.state.newValue}
+              />
+            </div>
+            <div>
```

Comment:
The styling of this doesn't really tell the user that there's an error with the particular field. Could make it red at a minimum? `<TextField>` is capable of displaying quite nicely styled error messages: https://material-ui.com/components/text-fields/#validation